src/audio/rtnr/rtnr.c

```diff
@@ -540,41 +515,57 @@ static int rtnr_copy(struct comp_dev *dev)
 	RTKMA_API_First_Copy(cd->rtk_agl, cd->source_rate, source->stream.channels);
 
 	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
-	frames = audio_stream_avail_frames(&source->stream, &sink->stream);
```

Comment:
as much as this is appreciated as this reduces the call stack, I don't think this i needed. We just need the control interface modified.

Reply:
@cujomalainey  Do you mean that we should use the original bypass path provided by RTNR, instead of introducing a new one?

Reply:
yea i don't see any issue with using that. the concern was the TLV interface only.

Reply:
Force-pushed a new commit. Please check it, thanks.